### DISCUSSION TITLE
Change JSON payload structure

### DISCUSSION THREAD

**Discussion by jarvissd** [arendst/Tasmota#20999] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
- I want to know how tasmota firmware gets state of relay, then creates a Json payload like: MQT: stat/ESP321/POWER:{"POWER":"ON"} . I don't know how to change the structure of JSON payload, i don't want to use Rule . Plss, help me. Thanks a lot!!!

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
There are no options to arbitrarily change payload structure, simplest is to use rules (whether you want it or not) to publish *additional* payloads.

**Reply by jarvissd** [reply]
i found i18n.h where i can change name of JSON payload but i could not find the way a JSON payload structure be created

**Reply by sfromis** [reply]
When you have a translation string name, you can search the code for where it is used.

**Comment by jarvissd** [comment]
@sfromis  i think it can be change inside tasmota code

**Reply by Jason2866** [reply]
Everything can be changed by modifying the code. You walk alone by doing this.
The way it can be achivied is already suggested -> rules.
